worker/provisioner/provisioner_task.go

```diff
@@ -1026,8 +1018,6 @@ func (task *provisionerTask) setupToStartMachine(machine apiprovisioner.MachineP
 		return environs.StartInstanceParams{}, errors.Annotatef(err, "creating instance config for machine %q", machine)
 	}
 
-	assocProvInfoAndMachCfg(pInfo, instanceCfg)
```

Comment:
Notice here that there is no assignment of the call return. It effectively does nothing.